refactor(forms): improve discoverability of ValidationError flavors

mmalerba opened the pull request:
Improves discoverability by putting the WithField, WithoutField, etc as subtypes of the main ValidationError type

mmalerba committed:
refactor(forms): improve discoverability of ValidationError flavors

Improves discoverability by putting the WithField, WithoutField, etc as
subtypes of the main ValidationError type

devversion reviewed (approved):
Reviewed-for: public-api

kirjs reviewed (approved):
reviewed-for: public-api

mmalerba committed:
docs: improve docs generation for namespaces

Improves docs epxerience for interfaces / type aliases that are merged
with a namespace.

JeanMeche committed:
fixup! docs: improve docs generation for namespaces
